tests/core/engine_adapter/test_spark.py

```diff
@@ -270,9 +270,10 @@ def test_replace_query_self_ref_not_exists(
 
     assert to_sql_calls(adapter) == [
         "CREATE TABLE IF NOT EXISTS `db`.`table` (`col` INT)",
-        # This second query doesn't do anything but we can't just run this because we need the table
-        # to exist since it is self-referencing
-        "CREATE TABLE IF NOT EXISTS `db`.`table` AS SELECT `col` + 1 AS `col` FROM `db`.`table`",
```

Comment:
This old behavior looks like a bug.

Reply:
It is actually an artifact of the test itself. There is a scenario the test took where it said the table does not exist when it has to since the SQL statement above created it. So the test was updated to dynamically determine if the table exists based on if a create table command was run and now the output is correct. 

TL;DR: Test bug not an actual behavior bug.